doc/lua_api.md

```diff
@@ -6562,6 +6566,14 @@ Formspec
         * `"INV"`: something failed
         * `"CHG"`: has been changed
         * `"VAL"`: not changed
+* `minetest.show_death_screen(player, reason)`
```

Comment:
`show_death_screen` instead of `show_death_formspec` because a new GUI API is planned that could be used here too